qiskit/algorithms/evolvers/variational/var_qte.py

```diff
@@ -0,0 +1,220 @@
+        self._allowed_num_instability_error = allowed_num_instability_error
+
+    def _evolve_helper(
+        self,
+        init_state_param_dict: Dict[Parameter, Union[float, complex]],
+        hamiltonian: OperatorBase,
+        time: float,
+        t_param: Parameter,
+        error_calculator=None,  # TODO will be supported in another PR
+        initial_state: Optional[Union[OperatorBase, QuantumCircuit]] = None,
+    ) -> OperatorBase:
+        r"""
+        Helper method for performing time evolution. Works both for imaginary and real case.
+
+        Args:
+            init_state_param_dict: Parameter dictionary with initial values for a given
+                parametrized state/ansatz.
+            hamiltonian:
+                Operator used for Variational Quantum Imaginary Time Evolution (VarQTE)
+                The coefficient of the operator (``operator.coeff``) determines the evolution
+                time.
+                The operator may be given either as a composed op consisting of a Hermitian
+                observable and a ``CircuitStateFn`` or a ``ListOp`` of a ``CircuitStateFn`` with a
+                ``ComboFn``.
+                The latter case enables the evaluation of a Quantum Natural Gradient.
+            time: Total time of evolution.
+            initial_state: Quantum state to be evolved.
+            t_param: Time parameter in case of a time-dependent Hamiltonian.
+
+        Returns:
+            Result of the evolution which is a quantum circuit with bound parameters as an
+            evolved state.
+        """
+
+        init_state_parameters = list(init_state_param_dict.keys())
+        init_state_parameters_values = list(init_state_param_dict.values())
+
+        metric_tensor = self._variational_principle.calc_metric_tensor(
+            initial_state, init_state_parameters
+        )
+        evolution_grad = self._variational_principle.calc_evolution_grad(
+            hamiltonian, initial_state, init_state_parameters
+        )
+
+        linear_solver = VarQTELinearSolver(
+            metric_tensor,
+            evolution_grad,
+            self._lse_solver_callable,
+            self._grad_circ_sampler,
+            self._metric_circ_sampler,
+            self._energy_sampler,
+            self._allowed_imaginary_part,
+        )
+
+        # Convert the operator that holds the Hamiltonian and ansatz into a NaturalGradient operator
+        self._ode_function_generator._lazy_init(
+            linear_solver,
+            error_calculator,
+            t_param,
+            init_state_param_dict,
+        )
+
+        ode_solver = VarQTEOdeSolver(
+            init_state_parameters_values, self._ode_function_generator, self._ode_solver_callable
+        )
+        parameter_values = ode_solver.run(time)
+        param_dict_from_ode = dict(zip(init_state_parameters, parameter_values))
+
+        return initial_state.assign_parameters(param_dict_from_ode)
+
+    def bind_initial_state(
```

Comment:
Can we rename this to `bind_parameters_to_state`? This is a useful function not only to bind the initial state parameters but for binding the parameters at an arbitrary time of the evolution to the ansatz.